workflows/differentialabundance.nf

```diff
@@ -157,8 +171,44 @@ workflow DIFFERENTIALABUNDANCE {
         ch_versions = ch_versions
             .mix(AFFY_JUSTRMA_RAW.out.versions)
 
-    }
-    else if(params.study_type == 'geo_soft_file'){
+    } else if (params.study_type == 'maxquant'){
+        // For maxquant, run proteus module to import the protein abundances
+
+        // Save contrasts file to channel
+        ch_contrasts_file = Channel.from([[exp_meta, file(params.contrasts)]])
+
+        // Split contrasts for proteus and for the later modules
+        ch_contrasts_split = ch_contrasts_file
+            .splitCsv ( header:true, sep:(params.contrasts.endsWith('tsv') ? '\t' : ','))
+            .map{ it.tail().first() }
+
+        // For proteus, extract only meta and contrast variable
+        ch_contrasts_proteus = ch_contrasts_split
+            .map{
+                tuple(
+                    exp_meta,       // meta map
+                    it.variable     // contrast variable
+                )
+            }
+
+        // For the plotting and following modules, save all contrast info
+        ch_contrasts = ch_contrasts_split
+            .map{
+                it.blocking = it.blocking.replace('NA', '')
+                if (!it.id){
+                    it.id = it.values().join('_')
+                }
+                tuple(it, it.variable, it.reference, it.target)
+            }
```

Comment:
```suggestion
```

Do you need any of this? `ch_contrasts` is redefined before LIMMA anyway, based on the VALIDATOR output

Reply:
Afraid so; PROTEUS needs the contrast variable, and in turn, the VALIDATOR needs PROTEUS output

Reply:
I think you're missing something- this definition of `ch_contrasts` is never used. It's redefined [here](https://github.com/nf-core/differentialabundance/blob/4d876af4756a5b2822fc31235ef16021e8db77c6/workflows/differentialabundance.nf#L325).

Reply:
You are right...I removed that